regulation-worker/cmd/main.go

```diff
@@ -21,15 +21,21 @@ import (
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/destination"
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/initialize"
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/service"
+	oauth "github.com/rudderlabs/rudder-server/router/oauthResponseHandler"
 	"github.com/rudderlabs/rudder-server/services/filemanager"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 )
 
 var pkgLogger = logger.NewLogger().Child("regulation-worker")
 
-func main() {
+func InitAll() {
 	initialize.Init()
 	backendconfig.Init()
+	oauth.Init()
+}
```

Comment:
I believe we shouldn't ideally be calling this function again anywhere else (even mistakenly). Hence, I recommend merging it with the `main` function for now.

Suppose the steps in the init phase grow big enough in the future. In that case, we can consider introducing a flag to track the initialization status and validate accordingly.

Reply:
If you keenly observe this is the same way that the `rudder-server` code is built. Not saying that this is the right way but I was being consistent with the approach we took inside the code-base, so that when we decide to apply changes, we do it in the same way across the code-base

Reply:
Please check [this](https://github.com/rudderlabs/rudder-server/blob/master/main.go#L141-L208)

Reply:
Got it. Should we add the flag for extra safety? Thoughts?

Reply:
Don't think a flag is a necessity